deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/AmqpDeviceOperations.java

```diff
@@ -102,7 +102,7 @@ public void openLinks(Session session) throws IllegalArgumentException
      */
     public synchronized boolean isReceiverLinkTag(String name)
     {
-        return name != null && this.receiverLinkTag.equals(name);
+        return this.receiverLinkTag.equals(name);
```

Comment:
Q - is "this.receiverLinkTag" ever null?

Reply:
It can be, but receiverLinkTag will not be, so receiverLinkTag.equals(null) -> false when it is.

Reply:
Officially, 

> For any non-null reference value x, x.equals(null) should return false.